warehouse/email/__init__.py

```diff
@@ -33,12 +35,27 @@ def _compute_recipient(user, email):
 
 
 @tasks.task(bind=True, ignore_result=True, acks_late=True)
-def send_email(task, request, recipient, msg):
+def send_email(task, request, recipient, msg, user_id, ip_address):
     msg = EmailMessage(**msg)
     sender = request.find_service(IEmailSender)
+    from_address = parseaddr(
+        sender.sender if hasattr(sender, "sender") else sender._sender
+    )[1]
```

Comment:
I'm not sure what reason for the difference in the class signature for `SMTPEmailSender` and `SESEmailSender` are, but I think we should probably unify this as something each service provides, so we can just do something like:

```python
from_address = sender.from_address
```
(or just put that inline below)